examples/contrib/oed/gp_bayes_opt.py

```diff
@@ -33,7 +34,9 @@ def update_posterior(self, X, y):
         X = torch.cat([self.gpmodel.X, X])
         y = torch.cat([self.gpmodel.y, y])
         self.gpmodel.set_data(X, y)
-        self.gpmodel.optimize(loss=TraceEnum_ELBO(strict_enumeration_warning=False).differentiable_loss)
+        optimizer = torch.optim.Adam(self.gpmodel.parameters())
+        gp.util.train(self.gpmodel, optimizer,
+                      loss_fn=TraceEnum_ELBO(strict_enumeration_warning=False).differentiable_loss)
```

Comment:
@ae-foster I have just revised the GP module to make it easier to use. However, I notice that the `oed` example failed because of my change. I look over the example and it seems to me that this together with a similar one in `ab_test.py` are the only places to change. But the test is still failing :(. Now, when a gpmodel is trained, it keeps trained values for parameters (previously, calling `pyro.clear_param_store()` will clear everything). I suspect that it causes some issues for this example. What do you think about it?

Reply:
By the way, why did you use `loss=TraceEnum_ELBO(strict_enumeration_warning=False).differentiable_loss` instead of `loss=TraceEnum_ELBO(strict_enumeration_warning=False)`? This is another hint to me.